inject-generator/src/test/java/io/avaje/inject/generator/models/valid/provider/FactoryProvider.java

```diff
@@ -0,0 +1,23 @@
+package io.avaje.inject.generator.models.valid.provider;
```

Comment:
We need to move these classes into blackbox and add a test that asserts the behaviour.